### ISSUE TITLE
foreignKey element reordered such that resultant XML is no longer DTD-compliant

### ISSUE THREAD

**Issue by bstoots** [dbsteward/dbsteward#121]
I needed to use the compound foreignKey functionality and found that during compilation the element order is changed such that foreignKey is above all other child elements inside table.  This then causes the compiled XML to fail DTD validation.  See below for an example error:

```
ERROR    /tmp/dbsteward_validate_tb39gG:17290: element table: validity error : Element table content does not follow the DTD, expecting (tablePartition? , tableOption* , column+ , index* , constraint* , foreignKey* , grant* , rows?), got (foreignKey column column constraint grant grant grant )
```

**Comment by mariasql** [comment]
I have exactly the same issue - was there any fix or maybe someone is planning to fiix?

**Comment by bstoots** [comment]
Let me see what is involved.  I may be able to put together a PR for this issue.

**Comment by bstoots** [comment]
Created a stripped down definition exposing the issue: https://gist.github.com/bstoots/93108f8f82706dfaaebabc97f201e6f2

**Comment by nkiraly** [comment]
Fixed by #135
